model/api_endpoint_lb.go

```diff
@@ -107,10 +129,32 @@ func (e APIEndpointLB) Validate() error {
 		}
 	}
 
-	if e.ManageELB() && len(e.APIAccessAllowedSourceCIDRs) == 0 && len(e.SecurityGroupIds) == 0 {
+	if e.ClassicLoadBalancer() && e.ManageELB() && len(e.APIAccessAllowedSourceCIDRs) == 0 && len(e.SecurityGroupIds) == 0 {
 		return errors.New("either apiAccessAllowedSourceCIDRs or securityGroupIds must be present. Try not to explicitly empty apiAccessAllowedSourceCIDRs or set one or more securityGroupIDs")
 	}
 
+	if !e.NetworkLoadBalancer() && !e.ClassicLoadBalancer() {
+		return errors.New("load balancer type must be either 'classic' or 'network'")
+	}
+
+	if e.NetworkLoadBalancer() {
+		defaultRanges := DefaultCIDRRanges()
+
+		if len(e.SecurityGroupIds) > 0 {
+			return errors.New("cannot specify security group IDs for a network load balancer")
+		}
+
+		if len(e.APIAccessAllowedSourceCIDRs) != len(defaultRanges) {
+			return errors.New("cannot override apiAccessAllowedSourceCIDRs for a network load balancer")
+		}
+
+		for i, r := range defaultRanges {
+			if r != e.APIAccessAllowedSourceCIDRs[i] {
```

Comment:
I guess we should add an controller sg ingress corresponds to apiAccessAllowedSourceCIDRs also for an NLB.

Oherwise a public API endpoint backed by an NLB isn't accessible from the Internet?

If we go that way, allowing to customize CIDRs here would be nicer for higher security.

Reply:
@mumoshu It's not possible to attach Security Groups to NLBs; one way to think of NLB is that it's an Elastic IP address that happens to go to multiple instances, instead of just one. Everything else stays the same.

What we can do here is restrict the usage of NLB-backed API endpoints to private NLBs only. This way, we still get to use NLB-backed API endpoints in worker nodes (in the hopes of mitigating issues like #598, but prevent users from creating a publicly-accessible NLB.

Reply:
@danielfm Thx! I think I understand that. Therefore, I suggest to alter "controller" sg ingress to allow apiAccessAllowedSourceCIDRs for publicly-accessible NLBs, instead of trying to attach sgs to an NLB which isn't possible.

Reply:
Not sure I'm following. What you're suggesting is to append the `apiAccessAllowedSourceCIDRs` from each NLB-backed API endpoint to the controller SG ingress rules?

Reply:
Exactly. It would look like similarly to the controller sg ingress for a vpc cidr you've added, which I believe the intention is to allow worker nodes to acess k8s api via NLB.

Reply:
@danielfm , suggestion by @mumoshu will work because NLB preserves source IP address, therefore SG would be able to filter traffic

Reply:
Done.